Mathlib/Data/Bitvec/Defs.lean

```diff
@@ -1,339 +1,56 @@
 /-
-Copyright (c) 2015 Joe Hendrix. All rights reserved.
+Copyright (c) 2023 Wojciech Nawrocki. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
-Authors: Joe Hendrix, Sebastian Ullrich
+Authors: Wojciech Nawrocki, Joe Hendrix, Harun Khan, Abdalrhman M Mohamed
 -/
-import Mathlib.Data.Vector.Basic
-import Mathlib.Data.Nat.Pow
-import Init.Data.Format.Basic
-import Mathlib.Init.Data.Nat.Lemmas
 
-#align_import data.bitvec.core from "leanprover-community/mathlib"@"1126441d6bccf98c81214a0780c73d499f6721fe"
-/-!
-# Basic operations on bitvectors
+import Mathlib.Data.Fin.Basic
+import Mathlib.Data.Nat.Bitwise
+import Mathlib.Data.ZMod.Defs
+import Std.Data.BitVec
 
-This is a work-in-progress, and contains additions to other theories.
 
-This file was moved to mathlib from core Lean in the switch to Lean 3.20.0c.
-It is not fully in compliance with mathlib style standards.
+/-!
+# Basic operations on bitvectors
+We define bitvectors. We choose the `Fin` representation over others for its relative efficiency
+(Lean has special support for `Nat`), alignment with `UIntXY` types which are also represented
+with `Fin`, and the fact that bitwise operations on `Fin` are already defined. Some other possible
+representations are `List Bool`, `{ l : List Bool // l.length = w}`, `Fin w → Bool`.
+
+We also define many bitvector operations from the
+[`QF_BV` logic](https://smtlib.cs.uiowa.edu/logics-all.shtml#QF_BV).
+of SMT-LIBv2.
 -/
 
-/-- `Bitvec n` is a `Vector` of `Bool` with length `n`. -/
-@[reducible]
-def Bitvec (n : ℕ) :=
-  Vector Bool n
-#align bitvec Bitvec
```

Comment:
Can you try to keep these aligns? For instance, this one would become `#align bitvec Std.BitVec`